lib/ansible/cli/galaxy.py

```diff
@@ -808,8 +808,6 @@ def parse_role_req(requirement):
             if "include" not in requirement:
                 role = RoleRequirement.role_yaml_parse(requirement)
                 display.vvv("found role %s in yaml file" % to_text(role))
-                if "name" not in role and "src" not in role:
-                    raise AnsibleError("Must specify name or src for role")
```

Comment:
I don't see that we do this same check elsewhere, but I would really move it to the `role_yaml_parse` above to either raise the error or just return a `None` role.

Reply:
I moved it to GalaxyRole rather than `role_yaml_parse`, because that method normalizes string requirements to always contain 'name' and 'src' (None by default), but GalaxyRole always requires a non-None name to install.

At runtime a role dependency with name `None` seems to just be ignored. If I move the error handling into `role_yaml_parse`, that would  change that, not sure what the right behavior is.

Reply:
> At runtime a role dependency with name None seems to just be ignored. If I move the error handling into role_yaml_parse, that would change that, not sure what the right behavior is.

Oops, I had a typo in the meta path so I didn't test this accurately initially.

This 
```
dependencies:
  - role: ~
```
is an error at runtime here
https://github.com/ansible/ansible/blob/devel/lib/ansible/playbook/role/definition.py#L123-L124

`dependencies: [""]` is ignored at runtime